### DISCUSSION TITLE
Run AltDrag elevated with no UAC prompt

### DISCUSSION THREAD

**Discussion by dogenzenji8** [stefansundin/altdrag#168] [Show and tell]
Thanks so much for creating and maintaining AltDrag @stefansundin! I use it so often that I was getting annoyed at AltDrag hanging after resuming from sleep on Windows 11, then having to manually kill and restart the app to use it again. I wanted to share something I discovered while coming up with a workaround. I settled on using Task Scheduler to execute `taskkill` then restart the app on resume from sleep when I noticed the option of "Run with highest privileges" which I promptly enabled. I then noticed that after executing this task, not only did it work, but AltDrag was running elevated without me ever having been prompted by UAC. So I decided I would always want AltDrag running elevated without being prompted so added a second trigger to the task to also execute on log in and configured AltDrag to not run on startup, but this didn't work, AltDrag always started in a hung state. I then changed AltDrag back to running on startup and configured the second task trigger to delay by one minute and viola! Now AltDrag runs elevated without UAC prompt whether I turn on, reboot, or resume my PC from sleep. Below is an XML export of the task which anyone can use after tweaking the AltDrag path and Windows user details. I hope you and others find this useful!

```
<?xml version="1.0" encoding="UTF-16"?>
<Task version="1.2" xmlns="http://schemas.microsoft.com/windows/2004/02/mit/task">
  <RegistrationInfo>
    <Date>2023-03-31T14:00:29.2416974</Date>
    <Author>[Your PC]\[Your User]</Author>
    <URI>\Microsoft\Windows\Restart AltDrag</URI>
  </RegistrationInfo>
  <Triggers>
    <EventTrigger>
      <Enabled>true</Enabled>
      <Subscription>&lt;QueryList&gt;&lt;Query Id="0" Path="System"&gt;&lt;Select Path="System"&gt;*[System[Provider[@Name='Microsoft-Windows-Power-Troubleshooter'] and EventID=1]]&lt;/Select&gt;&lt;/Query&gt;&lt;/QueryList&gt;</Subscription>
    </EventTrigger>
    <LogonTrigger>
      <Enabled>true</Enabled>
      <Delay>PT1M</Delay>
    </LogonTrigger>
  </Triggers>
  <Principals>
    <Principal id="Author">
      <UserId>[You user GUID]</UserId>
      <LogonType>InteractiveToken</LogonType>
      <RunLevel>HighestAvailable</RunLevel>
    </Principal>
  </Principals>
  <Settings>
    <MultipleInstancesPolicy>StopExisting</MultipleInstancesPolicy>
    <DisallowStartIfOnBatteries>false</DisallowStartIfOnBatteries>
    <StopIfGoingOnBatteries>true</StopIfGoingOnBatteries>
    <AllowHardTerminate>true</AllowHardTerminate>
    <StartWhenAvailable>false</StartWhenAvailable>
    <RunOnlyIfNetworkAvailable>false</RunOnlyIfNetworkAvailable>
    <IdleSettings>
      <StopOnIdleEnd>true</StopOnIdleEnd>
      <RestartOnIdle>false</RestartOnIdle>
    </IdleSettings>
    <AllowStartOnDemand>true</AllowStartOnDemand>
    <Enabled>true</Enabled>
    <Hidden>false</Hidden>
    <RunOnlyIfIdle>false</RunOnlyIfIdle>
    <WakeToRun>false</WakeToRun>
    <ExecutionTimeLimit>PT0S</ExecutionTimeLimit>
    <Priority>7</Priority>
  </Settings>
  <Actions Context="Author">
    <Exec>
      <Command>taskkill</Command>
      <Arguments>/f /im AltDrag.exe</Arguments>
    </Exec>
    <Exec>
      <Command>"C:\Users\[Your user]\AppData\Roaming\AltDrag\AltDrag.exe"</Command>
    </Exec>
  </Actions>
</Task>
```

**Comment by RamonUnch** [comment]
Nice trick indeed, however keep in mind that there have been a PR to fix the hang on sleep/resume and DPI changes #133. You might be interested in applying the said patch for yourself.
